sdk/core/azure-core/inc/azure/core/case_insensitive_map.hpp

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ * @brief A `map<string, T>` with case-insensitive key comparison.
+ */
+
+#pragma once
+
+#include "azure/core/internal/strings.hpp"
+
+#include <map>
+#include <string>
+
+namespace Azure { namespace Core {
+
+  /**
+   * @brief A type alias of `std::map<std::string, T>` with case-insensitive key comparison.
+   *
+   * @tparam T Type of values being stored in the map.
+   */
+  template <typename T>
```

Comment:
when T is not `std::string`?

Reply:
Never yet. I am fine to drop it, let me know. We always can reintrocuce it, and make it to be a type alias of `CaseInsensitiveMap<T>`. But what would we call this map?

Reply:
Hmm, ok, I think I know. Let's remove T, and if we need it later, we will do `template<typename T = std::string>` to not break the existing code.

Reply:
@Jinming-Hu, thank you, good question, check it out now.